Cmake build of Apache Arrow fails using grpc's Findre2.cmake with "add_library cannot create imported target "RE2::re2" because another target with the same name already exists."

The `grpc` cmake file [Findre2.cmake](https://github.com/grpc/grpc/blob/master/cmake/modules/Findre2.cmake) causes this cmake error when building downstream libraries that use both `grpc` and `re2`, specifically Apache [arrow](https://github.com/apache/arrow).

Error:
```
:info:build CMake Error at /opt/local/lib/cmake/grpc/modules/Findre2.cmake:29 (add_library):
:info:build   add_library cannot create imported target "re2::re2" because another target
:info:build   with the same name already exists.
```

Related:
* http://mail-archives.apache.org/mod_mbox/arrow-dev/202009.mbox/%3C6ECB7D60-844F-410D-8840-24F2CE261752@yahoo.com%3E
* http://mail-archives.apache.org/mod_mbox/arrow-jira/202009.mbox/%3CJIRA.13327607.1600156125000.372264.1600157460269@Atlassian.JIRA%3E
* https://www.mail-archive.com/dev@arrow.apache.org/msg20631.html
* https://github.com/macports/macports-ports/pull/7791

### What version of gRPC and what language are you using?

1.35.0

### What operating system (Linux, Windows,...) and version?

macOS 10.15.7 19H524
Xcode 12.4 12D4e 

Also observed on Linux; see above.

### What runtime / compiler are you using (e.g. python version or version of gcc)

cmake

### What did you do?

cmake build of Apache Arrow that uses `grpc`and `re2`  libraries.

See https://github.com/macports/macports-ports/blob/76927172cfe36acb7a6ea543223dd8849dbb0cb9/devel/apache-arrow/Portfile

See attached log for build details.

[main.log](https://github.com/grpc/grpc/files/6097304/main.log)

### What did you expect to see?

Successful build.

### What did you see instead?

The cmake error above.

--- a/cmake/modules/Findre2.cmake
+++ b/cmake/modules/Findre2.cmake
@@ -18,6 +18,13 @@ if(re2_FOUND)
   return()
 endif()
 
+# As per https://github.com/grpc/grpc/issues/25434, idempotence is necessary
+# because CMake fails when another target with the same name already exists.
+if(TARGET re2::re2)
+  message(STATUS "Found RE2 via pkg-config already?")
+  return()
+endif()
+
 find_package(PkgConfig REQUIRED)
 # TODO(junyer): Use the IMPORTED_TARGET option whenever CMake 3.6 (or newer)
 # becomes the minimum required: that will take care of the add_library() and

